logstash-core/src/main/java/org/logstash/ext/JrubyEventExtLibrary.java

```diff
@@ -234,21 +237,30 @@ public IRubyObject ruby_to_json(ThreadContext context, IRubyObject[] args)
             try {
                 return RubyString.newString(context.runtime, event.toJson());
             } catch (Exception e) {
-                throw RaiseException.from(context.runtime, RubyUtil.GENERATOR_ERROR, e.getMessage());
+                throw toRubyError(context, RubyUtil.GENERATOR_ERROR, e);
             }
         }
 
         // @param value [String] the json string. A json object/map will newFromRubyArray to an array containing a single Event.
         // and a json array will newFromRubyArray each element into individual Event
         // @return Array<Event> array of events
         @JRubyMethod(name = "from_json", required = 1, meta = true)
-        public static IRubyObject ruby_from_json(ThreadContext context, IRubyObject recv, RubyString value)
-        {
+        public static IRubyObject ruby_from_json(ThreadContext context, IRubyObject recv, RubyString value, final Block block) {
```

Comment:
slightly different that the Java `fromJson(final String json, final EventFactory factory)` signature.
the reason for that is the Ruby vs Java event duality, this slightly resembles the Java Event API (the EventFactory being a functional interface could be implemented as a block) while doing the Java wrapping/unwrapping seamlessly (as we start with `EventFactory` like implementations in Ruby).

*HINT: was wondering for a while now why `LogStash::Event` isn't just a `org.logstash.Event` to reduce the difference between the Java and Ruby API (something to consider for the maybe 8.0 future) - if this was considered while designing the Java event API (conceptually - the naming differences can be handled on the Ruby end by patching the Java class)*

Reply:
There was a lot of back-and-forth on this, but IIRC we explicitly avoided having any jruby-related infection of the Java Event API so that consumers of the API would have a much smaller surface area. The Ruby APIs are also _really_ hard to break confidently because we don't have good data on their usage.

Reply:
Usage of the Block and FunctionalInterface for the EventFactory is clever, and in a really nice way.

Reply:
Still consider the block-form to be a "hack" around needing to manually wrap returns from a Java `EventFactory`.
was trying to think about a potential future here: `LogStash::Event = org.logstash.Event` (whether it happens or not) in which case the block from would become (so we should consider it "internal").

> The Ruby APIs are also really hard to break confidently because we don't have good data on their usage.

I didn't mean breaking the Ruby API, Java API would have stayed as is and the extra Ruby behavior would be scripted on top `org.logstash.Event.class_eval { def get(arg); getField(arg); end }` etc.

From a user's perspective the duality is confusing.